### DISCUSSION TITLE
[SP7] [Arch] Is secureboot required for pen and touch?

### DISCUSSION THREAD

**Discussion by Droogfein** [linux-surface/linux-surface#1063] [Q&A]
Hi,

I was wondering if secureboot is required for the features to work. I do not want to risk bricking my device.
By following all the steps in the guide, except setting up secureboot, Pen and Touch still are not working, iptsd does not return any hidraw devices.

P.S.: Love the work that is being done SP is an amazing piece of hardware, I love that I don't have to get a seperate device to be able to use linux.
And I realize this might have been a better submission for Issues...

**Comment by StollD** [comment]
No, you dont need secureboot for the pen to work. All you need is the surface kernel and iptsd from the repository.

Could you please post your dmesg so that I can check what is going on?

```
$ sudo dmesg > dmesg.log
```

**Reply by Droogfein** [reply]
Certainly, here is the log. Thank you.
[dmesg.log](https://github.com/linux-surface/linux-surface/files/10776523/dmesg.log)

**Reply by StollD** [reply]
You are still booting the default Arch kernel, not the surface kernel (`uname -r` should contain surface).

**Reply by Droogfein** [reply]
That fixed it, thanks for the quick help. But why is there not a patch or dkms module, instead of the custom kernel? Are you waiting to get upstreamed?

**Reply by StollD** [reply]
Some of our patches cannot be installed through DKMS without needlessly copying an entire module. A custom kernel is more flexible and can be signed by us for secureboot. Especially on SP7, another module has to be patched for the touch driver to be loaded.

Regarding upstreaming, we are trying to have as much code as possible upstream, but it is difficult to say for the touch driver. There are a lot of subtile quirks on many devices, so without a spec and with only one (well-behaving) device the only thing I can do is push code and wait if it breaks on someones device. That is not a practice I consider suitable for upstream code. Another issue is that I can't promise to keep maintaining the driver once my surface dies.

I would very much welcome if Intel actually supported all of their weird technology upstream and not just the parts that they can sell to enterprise customers. And I don't want to follow their example of throwing code over the wall and running away from it.
